packages/core/src/odata/common/entity-deserializer.ts

```diff
@@ -146,23 +156,50 @@ export function entityDeserializer(edmToTs, extractODataETag) {
     }
   }
 
-  function deserializeComplexType<EntityT extends EntityBase>(
+  function deserializeComplexTypeLegacy<
+    EntityT extends EntityBase,
+    ComplexTypeNamespaceT extends ComplexTypeNamespace
+  >(
     json: MapType<any>,
-    complexTypeField: ComplexTypeField<EntityT>
+    complexTypeField: ComplexTypeField<EntityT, ComplexTypeNamespaceT>
   ): MapType<any> {
+    logger.warn(
+      'It seems that you are using an outdated OData client. To make this warning disappear, please regenerate your client using the latest version of the SAP Cloud SDK generator.'
+    );
     return Object.entries(complexTypeField)
       .filter(
         ([_, field]) =>
-          field instanceof EdmTypeField &&
+          (field instanceof EdmTypeField ||
+            field instanceof ComplexTypeField) &&
           typeof json[field._fieldName] !== 'undefined'
       )
-      .reduce((complexTypeObject, [fieldName, field]) => {
-        complexTypeObject[toPropertyFormat(fieldName)] = edmToTs(
-          json[field._fieldName],
-          field.edmType
-        );
-        return complexTypeObject;
-      }, {});
```

Comment:
That's how you fixed the legacy code, without tests 😼

Reply:
I added a test and currently this is tested and working correctly (The current v2 test services was not yet regenerated). Once I regenerate this won't be the case anymore.